addons/website/static/src/snippets/s_image_gallery/options.js

```diff
@@ -449,18 +453,44 @@ options.registry.GalleryImageList = options.registry.GalleryLayout.extend({
                 multiImages: true,
                 onlyImages: true,
                 save: images => {
+                    const imagePromises = [];
                     for (const image of images) {
-                        $('<img/>', {
+                        const $img = $('<img/>', {
                             class: $images.length > 0 ? $images[0].className : 'img img-fluid d-block ',
                             src: image.src,
                             'data-index': ++index,
                             alt: image.alt || '',
                             'data-name': _t('Image'),
                             style: $images.length > 0 ? $images[0].style.cssText : '',
                         }).appendTo($container);
+                        const imgEl = $img[0];
+                        imagePromises.push(new Promise(resolve => {
+                            loadImageInfo(imgEl, this._rpc.bind(this)).then(() => {
+                                if (imgEl.dataset.mimetype && ![
+                                    "image/gif",
+                                    "image/svg+xml",
+                                    "image/webp",
+                                ].includes(imgEl.dataset.mimetype)) {
+                                    // Convert to webp but keep original width.
+                                    imgEl.dataset.mimetype = "image/webp";
+                                    applyModifications(imgEl, {
+                                        mimetype: "image/webp",
+                                    }).then(src => {
+                                        imgEl.src = src;
+                                        imgEl.classList.add("o_modified_image_to_save");
+                                        resolve();
+                                    });
+                                } else {
+                                    resolve();
+                                }
+                            });
+                        }));
```

Comment:
Is there a reason why you do
```js
imagePromises.push(new Promise(resolve => {
    callAsyncFunc().then(() => {
        anotherCallToAnAsyncFunc().then(() => {
            aLastCallToAnAsyncFunc().then(() => {
                // Some code
                resolve();
...
savedPromise = Promise.all(imagePromises);
```
Cant you do something like that instead
```js
for (const image of images) {
    const a = await callAsyncFunc();
    ...
    const b = await anotherCallToAnAsyncFunc();
    ...
    const c = await aLastCallToAnAsyncFunc();
```
This would be more readable. But maybe there is a reason why you don't do it ?

Reply:
This is to run the rpcs in parallel, and also because `save` is not `async` - thus preventing me from awaiting in stable.